client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -0,0 +1,422 @@
+		} );
+	},
+
+	renderItem( item ) {
+		const onChange = ( ...args ) => this.props.onChange( item, ...args );
+		const setItemRef = ( ...args ) => this.setItemRef( item, ...args );
+
+		const { multiple, defaultTermId, translate } = this.props;
+		const selectedIds = this.getSelectedIds();
+		const itemId = item.ID;
+		const name = unescapeString( item.name ) || translate( 'Untitled' );
+		const checked = includes( selectedIds, itemId );
+		const inputType = multiple ? 'checkbox' : 'radio';
+		let disabled;
+
+		if ( multiple && checked && defaultTermId &&
+				( 1 === selectedIds.length ) &&
+				( defaultTermId === itemId ) ) {
+			disabled = true;
+		}
+
+		const input = (
+			<input
+				type={ inputType }
+				value={ itemId }
+				onChange={ onChange }
+				disabled={ disabled }
+				checked={ checked }
+			/>
+		);
+
+		return (
+			<div
+				key={ itemId }
+				ref={ setItemRef }
+				className="term-tree-selector__list-item">
+				<label>
+					{ input }
+					<span className="term-tree-selector__label">{ name }</span>
+				</label>
+				{ item.items && (
+					<div className="term-tree-selector__nested-list">
+						{ item.items.map( this.renderItem ) }
+					</div>
+				) }
+			</div>
+		);
+	},
+
+	renderRow( { index } ) {
+		if ( this.hasNoSearchResults() || this.hasNoTerms() ) {
+			return (
+				<div key="no-results" className="term-tree-selector__list-item is-empty">
+					{ ( this.hasNoSearchResults() || ! this.props.emptyMessage ) && (
+						<NoResults
+							createLink={ this.props.createLink } />
+					) }
+					{ this.hasNoTerms() && this.props.emptyMessage }
+				</div>
+			);
+		}
+
+		const item = this.getItem( index );
+		if ( item ) {
+			return this.renderItem( item );
+		}
+
+		if ( ! this.props.loading && this.props.lastPage ) {
+			return;
+		}
+
+		return (
+			<div key="placeholder" className="term-tree-selector__list-item is-placeholder">
+				<label>
+					<input
+						type={ this.props.multiple ? 'checkbox' : 'radio' }
+						disabled
+						className="term-tree-selector__input" />
+					<span className="term-tree-selector__label">
+						{ this.translate( 'Loading…' ) }
+					</span>
+				</label>
+			</div>
+		);
+	},
+
+	render() {
+		const rowCount = this.getRowCount();
+		const isCompact = this.isCompact();
+		const showSearch = this.state.searchTerm.length > 0 || ! isCompact;
+		const { className, loading, siteId, taxonomy, query } = this.props;
+		const classes = classNames( 'term-tree-selector', className, {
+			'is-loading': loading,
+			'is-compact': isCompact
+		} );
+
+		return (
+			<div ref={ this.setSelectorRef } className={ classes }>
+				{ this.state.requestedPages.map( ( page ) => (
+					<QueryTerms
+						key={ `query-${ page }` }
+						siteId={ siteId }
+						taxonomy={ taxonomy }
+						query={ { ...query, page } } />
+				) ) }
+				{ showSearch && (
+					<Search
+						searchTerm={ this.state.searchTerm }
+						onSearch={ this.onSearch } />
+				) }
+				<InfiniteLoader
```

Comment:
Why do you still need `<InfiniteLoader />` with `VirtualScroll`'s `onRowsRendered` handling setting pages to be requested?